### DISCUSSION TITLE
Programatic way to check if a property is build time

### DISCUSSION THREAD

**Discussion by shawkins** [quarkusio/quarkus#38173] [Community]
Keycloak for several reasons captures whether a given option is build time or not - however after adding some stricter logic about the handling of build time properties we've noticed several of our options were marked incorrectly as run time. If we have a set of quarkus property names, is there a way that we can programatically check which ones are build time? 

see https://github.com/keycloak/keycloak/issues/26175

**Comment by geoand** [comment]
There is nothing ready made OOTB, but it is theoretically possible (a good idea of how it could be done it follow the logic of how Quarkus [warns](https://github.com/quarkusio/quarkus/blob/3.6.6/core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigRecorder.java#L66) about a build time property being overridden at runtime).

In any case, cc @radcortez

**Comment by radcortez** [comment]
Yes, it is possible.

Check https://github.com/quarkusio/quarkus/blob/9dd3919464e1135599faa3764f158014b68fd868/core/deployment/src/main/java/io/quarkus/deployment/builditem/ConfigurationBuildItem.java#L10

And then:
https://github.com/quarkusio/quarkus/blob/6053da3880cdbb5a53092b2c320be89cad77f7a8/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java#L1221

**Reply by shawkins** [reply]
Thank you @radcortez let me see how far I can get with that.
